### PR TITLE
Add guide on alternative migration method for thanos to mimir

### PR DESCRIPTION
**Pull request by gmintoco**
#### What this PR does

Adds a link to the following [guide ](https://gist.github.com/gmintoco/e949cb9bf12e427449b36d6897239bcf) that describes a method of linking Mimir behind Thanos to allow users to query historical thanos data that may have been downsampled during the transition period.

This should probably be covered behind some warnings saying that it is unsupported or something to that effect.

This is my first stab at this to see if there is interest in this documentation. I would be happy to flesh this out a little more if it appears to be worth adding. Please let me know any feedback you might have.

#### Checklist

- [ ] Tests updated
- [ x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Comment by osg-grafana** [comment]
Thank you for taking a stab at docs! :) I will hold off on a doc review until feedback comes in from an engineer wrt including or not including this information.

**Comment by pracucci** [comment]
> Was this your understanding?

No, it wasn't, sorry!

> Instead we deploy a thanos-sidecar alongside mimir and “trick” the thanos-sidecar into thinking that mimir is a normal Prometheus server.

I took a better look at your nginx config and now I've a better understanding of it. Since you're just using the Prometheus-compatible API exposed by Mimir, then it should work in the future too.

File: docs/sources/migration-guide/migrating-from-thanos-or-prometheus.md
Comment (pracucci) [inline docs/sources/migration-guide/migrating-from-thanos-or-prometheus.md:71]:
```suggestion
An alternative to migrating the blocks is to continue to run Thanos and Mimir in parallel. You can direct clients (ie. Grafana) to Thanos and then setup Thanos to query Mimir via the Prometheus-compatible API endpoints exposed by Mimir. This allows you to query historical data from the Thanos store-gateway along with most recent data being pushed into Mimir from your Prometheus servers. You will need to carefully consider the external labels Mimir will present to Thanos along with the external labels that currently exist in Thanos to ensure compatibility with existing dashboards. Once the Thanos retention period has passed you can then tranistion all clients to query Grafana Mimir directly. This approach is very much experimental and has not yet seen significant use in production. For a write-up please see this link [here](https://gist.github.com/gmintoco/e949cb9bf12e427449b36d6897239bcf).
```

Comment (pracucci) [inline docs/sources/migration-guide/migrating-from-thanos-or-prometheus.md:71]: I'm not sure this is right place in the doc to add your method, cause this section is "Considerations on Thanos specific features" and I'm not sure an alternative migration path fits well into it.

Would you be open to actually create a new doc page (e.g. "Migrating from Thanos to Mimir using Thanos sidecar") and actually importing all your gist content into the new page? In the new doc page you could state it's experimental and not battle tested.

**Review by pracucci — COMMENTED** [review]
Thanks! I left a comment

**Comment by replay** [comment]
The CHANGELOG has just been cut to prepare for the next Mimir release. Please rebase `main` and eventually move the CHANGELOG entry added / updated in this PR to the top of the CHANGELOG document. Thanks!

File: docs/sources/migration-guide/migrating-from-thanos-or-prometheus.md
Comment (gmintoco) [reply docs/sources/migration-guide/migrating-from-thanos-or-prometheus.md:71]: yep that makes sense

**Review by gmintoco — COMMENTED** [review]
will work on a new document incorporating the gist as described

**Comment by gmintoco** [comment]
added a new doc with structure as requested. removed the changes from the old doc. Let me know if you have any further feedback :)

File: docs/sources/migration-guide/migrating-from-thanos-to-mimir-with-thanos-sidecar.md
Comment (osg-grafana) [inline docs/sources/migration-guide/migrating-from-thanos-to-mimir-with-thanos-sidecar.md:3]:
```suggestion
title: "Migrate from Thanos to Mimir using a Thanos sidecar"
menuTitle: "Migrate from Thanos to Mimir using a Thanos sidecar"
```

File: docs/sources/migration-guide/migrating-from-thanos-to-mimir-with-thanos-sidecar.md
Comment (osg-grafana) [inline docs/sources/migration-guide/migrating-from-thanos-to-mimir-with-thanos-sidecar.md:4]:
```suggestion
description: "Learn how to migrate from Thanos to Mimir using Thanos sidecar"
```

File: docs/sources/migration-guide/migrating-from-thanos-to-mimir-with-thanos-sidecar.md
Comment (osg-grafana) [inline docs/sources/migration-guide/migrating-from-thanos-to-mimir-with-thanos-sidecar.md:10]:
```suggestion
# Migrate from Thanos to Mimir using Thanos sidecar

As an operator, you can migrate a deployment of Thanos to Grafana Mimir by using Thanos sidecar to move your metrics over with a stepped workflow.
```

**Comment by gmintoco** [comment]
@osg-grafana Changes have been implemented as requested

**Comment by osg-grafana** [comment]
Thank you, @pstibrany. I ran into issues trying to run that because it was confusing my two GitHub accounts.

**Review by osg-grafana — APPROVED** [review]

**Comment by osg-grafana** [comment]
I can perform an edit in a subsequent PR, but let’s get this in before we start seeing merge conflicts based on upcoming work that I will perform.

**Review by pstibrany — APPROVED** [review]

**Pull request merged**
